tests/test_to_number.cc

```diff
@@ -87,6 +87,12 @@ TEMPLATE_TEST_CASE("test details: error_in_ulp()", "[to_number]",
 
 TEST_CASE("to_number(): Integer types", "[to_number]")
 {
+    REQUIRE(to_number<bool>("0").value() == false);
+    REQUIRE(to_number<bool>("1").value() == true);
+    REQUIRE(to_number<bool>("TRUE").value() == true);
+    REQUIRE(to_number<bool>("false").value() == false);
```

Comment:
We could maybe add a few more variations of upper- and lowercase here to make sure that the function does what it is supposed to.

Reply:
Also tests for the likes of `"TRUELIES"` and `"000"` etc are missing if I'm not mistaken.

And here you can argue, if `"000"` not also `false`?
`to_number<int>("00000") == 0`.

This (the limitation with `bool` to `"0"` and `"1"`) is somehow not consistent.